src/lib/utils/retryImport.web.js

```diff
@@ -1,6 +1,7 @@
 // @flow
 import { retry } from './async'
+import restart from './restart'
 
-const retryImport = fn => retry(fn, 5, 1000)
+const retryImport = fn => retry(fn, 5, 1000).catch(() => restart())
```

Comment:
I checked how we're updating the app via serviceWorker so i could suggest now the following
1. check the error object
2. it it's not the 404 chunk error - re-throw it
3. it it's 404 exactly - return never ending promise

Something like this one:

```javascript
.catch(e) => {
  if (!is404ChunkError(e)) {
    throw e
  }

  return new Promise(noop)
}

In the case of app updated but the old index.html was cached, user won't see any error
He will still see an loading screen until `useUpgradeDialog()` will receive update event from service worker.
After that user will see our generic new version available dialog, after closing which the app will be restarted with all cache cleaned